packages/sado-connect/src/providers/SadoContext.tsx

```diff
@@ -70,18 +85,21 @@ const NETWORK = "network";
 // Helper function to get item from sessionStorage
 function getItemFromSessionStorage<T>(key: string): T | null {
   try {
-    return sessionStorage.getItem(key) as T | null;
+    return JSON.parse(sessionStorage.getItem(key)) as T | null;
   } catch (error) {
     console.error(`Error retrieving ${key} from sessionStorage`, error);
     return null;
   }
 }
 
 // Helper function to set item to sessionStorage
-function setItemToSessionStorage(key: string, value: string | null) {
+function setItemToSessionStorage(
+  key: string,
+  value: string | null | BiAddress<any>,
+) {
   try {
     if (value) {
-      sessionStorage.setItem(key, value);
+      sessionStorage.setItem(key, JSON.stringify(value));
     } else {
       sessionStorage.removeItem(key);
     }
```

Comment:
this is not backward compatible, it will break when local storage retrieved non-json format.

Reply:
Could not reproduce (will KIV)